Feature validation

slifty opened the pull request:
All universal data elements are now validated according to the validation rules laid out in http://www.hudhdx.info/Resources/Vendors/4_0/HMISCSVSpecifications4_0FINAL.pdf

Error objects for validation reflect the HUD field as well as provide a message about what is incorrect.

slifty committed:
Fixed errors to render in XML

slifty committed:
Merge branch 'feature-configuration-refactor' of github.com:PCNI/OpenHMIS into hotfix-xml

slifty committed:
Added wrappers around good results

slifty committed:
Merge branch 'feature-configuration-refactor' of github.com:PCNI/OpenHMIS into hotfix-xml

slifty committed:
XML now returns codes not strings for fields

slifty committed:
Example validation for SSN and DoB

slifty committed:
Invalid enum values throw errors

slifty committed:
added logging to all exceptions

slifty committed:
Added XMLSeeAlso patch to single item DTO as well

slifty committed:
Merge branch 'master' of github.com:PCNI/OpenHMIS into feature-validation

slifty committed:
Merge branch 'hotfix-xml' of github.com:PCNI/OpenHMIS into feature-validation

slifty committed:
Client fields are now validated

slifty committed:
Universal Enrollment fields are now validated

slifty committed:
All universal data elements are now validated

slifty committed:
zomg that configuration setting can't come soon enough

cecilia-donnelly commented on src/main/java/org/openhmis/provider/LoggerExceptionMapper.java:
```diff
@@ -1,25 +1,25 @@
+//                   .build();
+//        }
+//    }
+//}
```
@slifty, guessing this goes with 740189b?

Commit:
Return logging removed in 1b3de81d

I think this logging should still be here.  Is that right, @slifty?

Commit:
Only validate existing SSNs

Don't check the format of null SSNs.

Commit:
Change example POST call

Add "data" wrapper in example GET call, but that still needs to be
expanded on.

Commit:
Account for null values in veteran history

This could use some review from @slifty, but it should fix #52.  I can
now post null values for these fields, save the new client correctly,
and get a 200 response as I should.

Commit:
Account for more null values: #52

We may need to change the solution in this and b8d01da to something
else, but all of the fields in these two commits need to be able to
handle nulls in POST requests, and now they can.
